tidy3d/components/tcad/simulation/heat_charge.py

```diff
@@ -81,6 +81,7 @@
     plot_params_heat_source,
 )
 from tidy3d.components.types import TYPE_TAG_STR, Ax, Bound, ScalarSymmetry, Shapely, annotate_type
+from tidy3d.components.utils import shape_swap_xy
```

Comment:
I visually verified that following examples work with `transpose=True`:
- [HeatSolver.ipynb](https://github.com/flexcompute/tidy3d-notebooks/blob/develop/HeatSolver.ipynb),
- [HeatDissipationSOI.ipynb](https://github.com/flexcompute/tidy3d-notebooks/blob/develop/HeatDissipationSOI.ipynb)
- [ThermallyTunedRingResonator.ipynb](https://github.com/flexcompute/tidy3d-notebooks/blob/develop/ThermallyTunedRingResonator.ipynb)
- [MachZehnderModulator.ipynb](https://github.com/flexcompute/tidy3d-notebooks/blob/develop/MachZehnderModulator.ipynb),
- [ChargeSolver.ipynb](https://github.com/flexcompute/tidy3d-notebooks/blob/develop/ChargeSolver.ipynb),